app/middleware/api.js

```diff
@@ -1,9 +1,21 @@
+function sanitize_username(username) {
```

Comment:
I'm happy with having this in a function and obviously good to lowercase the username...although, not so sure about the other transformations, doesn't this mean that **potentially** we make a request to a wrong namespace?

Reply:
[Kubernetes namespaces have to be valid DNS labels](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/#namespaces-and-dns). This is the same way we santize URLs everywhere else (or it should be, if not).

Reply:
Technically, we don't have to strip the non-alphanumeric character from the start of the username (L3), because the namespace is always prefixed with `user-`. Also, trimming to 63 chars doesn't take the `user-` prefix into account. I'll have another go at this.